### PR TITLE
feat: eth in deposit page's currency list

### PR DESCRIPTION
**Pull request by chrisling-dev**
https://www.notion.so/prepo/6ff94d2f628847cd97c80bfbe667b5b9?v=6b4403e1ecc748a48a63e32a1f35e012&p=d3eb439e1bd343cc96e906d26fb78258&pm=s

### CODE REVIEW

File: apps/frontend/core-dapp/src/stores/TokensStore.ts [hunk 1]
```diff
@@ -77,4 +91,21 @@ export class TokensStore {
...
+      if (aBalance === undefined || bBalance === undefined) return 0
+      return +bBalance - +aBalance
+    })
+  }
+
+  get defaultDepositToken(): Token | undefined {
+    const loadingIndex = this.sortedDepositTokens.findIndex(
+      (token) => this.getTokenBalance(token) === undefined
+    )
+    if (loadingIndex >= 0) return undefined
```
Comment (chrisling-dev) [inline apps/frontend/core-dapp/src/stores/TokensStore.ts:108]: returns undefined if any of the token's balance is still loading

**Review by chrisling-dev — COMMENTED** [review]
self reviewed

**Comment by MeLlamoPablo** [comment]
[The insufficientBalance check still looks for the base token](https://github.com/prepo-io/prepo-monorepo/pull/934/files#diff-9dd92de77048b5e769c3dfcf576ddd3f256c31615ef4836d975aba29bec8f6f1R153-R154) so it is possible to have enough balance and get the "Insufficient Balance" state:

Screenshot 2023-02-23 at 09 22 32

**Comment by chrisling-dev** [comment]
> [The insufficientBalance check still looks for the base token](https://github.com/prepo-io/prepo-monorepo/pull/934/files#diff-9dd92de77048b5e769c3dfcf576ddd3f256c31615ef4836d975aba29bec8f6f1R153-R154) so it is possible to have enough balance and get the "Insufficient Balance" state:
> 
> Screenshot 2023-02-23 at 09 22 32

yeap mentioned in discord that this will be fixed in separate PR because this PR has gotten quite big

**Comment by MeLlamoPablo** [comment]
> yeap mentioned in discord that this will be fixed in separate PR because this PR has gotten quite big

Oh sorry, didn't make that connection.

**Comment by chrisling-dev** [comment]
> > yeap mentioned in discord that this will be fixed in separate PR because this PR has gotten quite big
> 
> Oh sorry, didn't make that connection.

np!

**Review by MeLlamoPablo — APPROVED** [review]
The code LGTM

I can't say much about the actual UX since it's not currently testable, so I'll probably do more in-depth testing in the next PR

**Comment by chrisling-dev** [comment]
> 

thanks! the UX to test in this PR would be the selecting of ETH vs WETH and that the balances are correct, everything else would go to separate PRs

**Pull request merged**